pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -316,20 +316,20 @@ func validateImmutableFieldsCluster(new, old *Cluster) field.ErrorList {
 
 	clusterlog.Info("Cluster config is associated with management cluster", "name", old.Name)
 
-	if !RefSliceEqual(new.Spec.IdentityProviderRefs, old.Spec.IdentityProviderRefs) {
+	if !RefSliceEqual(new.Spec.IdentityProviderRefs, old.Spec.IdentityProviderRefs) || old.ExperimentalSelfManagedClusterUpgrade() {
```

Comment:
do we need this here? it seems like the method will return if old.ExperimentalSelfManagedClusterUpgrade() is true on line 314